src/test/kotlin/ua/marchenko/artauction/auction/repository/AuctionRepositoryTest.kt

```diff
@@ -36,102 +40,136 @@ class AuctionRepositoryTest : AbstractBaseIntegrationTest {
         val savedAuction = auctionRepository.save(auction)
 
         // THEN
-        assertEquals(auction.copy(id = savedAuction.id), savedAuction)
+        savedAuction.test()
+            .assertNext { auctionFromMono -> assertEquals(auction.copy(id = auctionFromMono.id), auctionFromMono) }
+            .verifyComplete()
     }
 
     @Test
     fun `should find auction by id when auction with this id exists`() {
         // GIVEN
-        val savedAuction = auctionRepository.save(MongoAuction.random(id = null))
+        val savedAuction = auctionRepository.save(
+            MongoAuction.random(
+                id = null,
+                startedAt = LocalDateTime.now().truncatedTo(ChronoUnit.MILLIS),
+                finishedAt = LocalDateTime.now().plusDays(1).truncatedTo(ChronoUnit.MILLIS)
+            )
+        ).block()
 
         // WHEN
-        val result = auctionRepository.findById(savedAuction.id.toString())
+        val result = auctionRepository.findById(savedAuction?.id.toString())
 
         // THEN
-        assertEquals(savedAuction.id, result?.id)
+        result.test()
+            .expectNext(savedAuction)
+            .verifyComplete()
     }
 
     @Test
-    fun `should return null when there is no auction with this id`() {
+    fun `should return empty when there is no auction with this id`() {
         // WHEN
         val result = auctionRepository.findById(ObjectId().toString())
 
         // THEN
-        assertNull(result, "Found auction must be null")
+        result.test()
+            .verifyComplete()
     }
 
     @Test
     fun `should return full auction with buyers and artwork when auction with this id exists`() {
         // GIVEN
-        val savedArtist = userRepository.save(MongoUser.random(id = null))
-        val savedArtwork = artworkRepository.save(MongoArtwork(artistId = savedArtist.id))
-        val savedBuyer = userRepository.save(MongoUser.random(id = null, role = Role.BUYER))
-        val buyers = listOf(
-            MongoAuction.Bid.random(buyerId = savedBuyer.id!!.toHexString()),
-        )
-        val savedAuction = auctionRepository.save(MongoAuction(buyers = buyers, artworkId = savedArtwork.id))
+        val savedArtist = userRepository.save(MongoUser.random(id = null)).block()
+        val savedArtworkFull = artworkRepository.save(MongoArtwork(artistId = savedArtist?.id))
+            .block()?.toFullArtwork(savedArtist)
```

Comment:
```suggestion
            .block()!!.toFullArtwork(savedArtist)
```

As for me, this is the case when we can use `!!` operator. So, the test would be cleaner